### ISSUE TITLE
Errors not catchable, "die();" is everywhere

### ISSUE THREAD

**Issue by trendoman** [CouchCMS/CouchCMS#154]
TLDR; Errors that invariably lead to 'die' may throw some event / exception at the very least, to intercept them in addons. If there is no action from the listener / handler → then die(); 

I realize the rewriting the whole codebase is not an option to replace errors with callbacks (and continue if error is 'skippable' as decided by the callback); but in the every coming released code there must be an organized effort to deal with errors!

- context may be provided by the addons (e.g., path to file, trace steps)
- a dedicated addon may be created to show tips and doc extracts on how to avoid the error
- multilanguage strings can be displayed instead of default English if admin-panel is set to non-english locale.

PHP developer must spend a lot of time to find all possible conditions where Couch dies to provide a meaningful context for a possible error. Example: Addon 'func-on-demand' may be configured to preload "<cms:func>" from user-defined locations; but if a user had registered a func via some other PHP code, there will be an uncatchable error, that does not provide any context or hint.

**Comment by trendoman** [comment]
Identically phrased error **Page not found** occurs at least 15 times throughout the code without additional information on various non-identical occasions. This makes error handling more complicated, even if intercepted via external tools. I add to the issue's message a request to ivariably add #ref numbers to the errors to clearly know where it comes from. In the future, we may create a reference book with details on those.
